app/helpers/application_helper/toolbar/middleware_domain_center.rb

```diff
@@ -16,4 +16,22 @@ class ApplicationHelper::Toolbar::MiddlewareDomainCenter < ApplicationHelper::To
                    ]
                  ),
                ])
+  button_group('middleware_server_operations', [
+                 select(
+                   :middleware_server_power_choice,
+                   'fa fa-power-off fa-lg',
+                   t = N_('Power'),
+                   t,
+                   :items => [
+                     button(
+                       :middleware_server_stop,
```

Comment:
Having `:middleware_server_stop` means that any user with privileges to stop a server will also have privileges to stop a whole domain. It's worth asking if that's how it should work or if a new privilege should be created.